crates/cheatcodes/src/inspector.rs

```diff
@@ -1602,6 +1622,13 @@ impl Cheatcodes {
                         None
                     };
 
+                    if let Some(ref mut zk_tx) = zk_tx {
+                        zk_tx.factory_deps = factory_deps.clone();
+                        Some(zk_tx)
+                    } else {
+                        None
```

Comment:
I'm also confused why is this else block here, it seems to be a noop.

Reply:
@Jrigada do we need this `else` block? Seems we don't save it in a variable and just drop it.

Reply:
It's not a noop but it is weird looking, I simplified this by putting it earlier.